offer-manager-domain/src/main/groovy/life/qbic/business/offers/Offer.groovy

```diff
@@ -447,11 +444,34 @@ class Offer {
     }
 
     private double calculateNetPrice() {
-        double netSum = 0.0
-        for (item in items) {
-            netSum += item.quantity * item.product.unitPrice
+        double netPrice
+        switch (selectedCustomerAffiliation.category) {
+            case AffiliationCategory.INTERNAL:
+                netPrice = calculateInternalNetPrice()
+                break
+            default:
+                netPrice = calculateExternalNetPrice()
+                break
```

Comment:
I think a simple if else would also be sufficient here. Switch makes imo only sense if you have a lot of cases.
Here it would then be if(selectedCustomerAffiliation.category.equals(AffiliationCategory.INTERNAL)) ... else ...

Reply:
it could be but in case we want to change that I would leave it as is. I also find it more readable

Reply:
I dont think this is going to change at any point ..